src/compile.js

```diff
@@ -2011,8 +2048,8 @@ Compiler.prototype.cclass = function (s) {
     scopename = this.enterScope(s.name, s, s.lineno);
     entryBlock = this.newBlock("class entry");
 
-    this.u.prefixCode = "var " + scopename + "=(function $" + s.name.v + "$class_outer($globals,$locals,$cell){var $gbl=$globals,$loc=$locals;";
-    this.u.switchCode += "(function $" + s.name.v + "$_closure(){";
+    this.u.prefixCode = "var " + scopename + "=(function $" + s.name.v + "$class_outer($globals,$locals,$cell){var $gbl=$globals,$loc=$locals;$free=$globals;";
+    this.u.switchCode += "(function $" + s.name.v + "$_closure($cell){";
```

Comment:
<3 for this fix
(TODO I've not fully reviewed this chunk but this looks very much like the fix to a bug I've seen intermittently for a while and would like to see chased down...)

Reply:
I think this was to fix nested closure's I think ... It's been a while.

Reply:
Yeah; that's the bug I spent too long banging my head against :)